### PR TITLE
Testing Tic Tac Toe Ruby code with Rspec

### PR DESCRIPTION
**Pull request by RaminMammadzada**
- Create specs for all public methods are created in Tic Tac Toe project.
- Wrap specs for each method in describe block, hints.
- Used let instead of before_each to define a variable for the test cases.
- Positive and negative scenarios are tested.
- Added testing section to README file for explanation

### CODE REVIEW

File: spec/game_spec.rb [hunk 1]
```diff
@@ -17,80 +17,173 @@
...
 
-  describe "#move" do
+  describe '#move' do
     it "should update the board after a player's move, we choose location 2" do
+      puts "For this special test, please enter '2' to select the location in the board !..."
       game.move(player1)
-      expect(game.board.board["2"]).to eql(player1.tag)
+      expect(game.board.board['2']).to eql(player1.tag)
     end
   end
```
Comment (bertil291utn) [inline spec/game_spec.rb]: ## Error running test

- Due to `gets.chomp` statement on game.rb file, it's displaying this error 
  - [ ] Remove on game.rb, this statement is only allowed on `main.rb`

image

File: spec/board_spec.rb [hunk 2]
```diff
@@ -22,87 +22,82 @@
     # end
   end
```
Comment (bertil291utn) [inline spec/board_spec.rb]: ## Remove inline comments

```ruby
# it "return name modified " do
    #   expect(board.player1).to eql(:player1)
    # end
```

File: spec/board_spec.rb [hunk 3]
```diff
@@ -22,87 +22,82 @@
...
-  describe "#update_board" do
-    it "update a board after modification" do
-      board.update_board('2',player1.name)
-      expect { board.print_board() }.to output(" |#{player1.tag}| \n-+-+-\n | | \n-+-+-\n | | \n").to_stdout
+  describe '#update_board' do
+    it 'update a board after modification' do
+      board.update_board('2', player1.name)
+      expect { board.print_board }.to output(" |#{player1.tag}| \n-+-+-\n | | \n-+-+-\n | | \n").to_stdout
       # expect(board.player2.name).to eql('zoro')
     end
```
Comment (bertil291utn) [inline spec/board_spec.rb]: ## Remove inline comments
```ruby
# expect(board.player2.name).to eql('zoro')
```

File: spec/player_spec.rb [hunk 4]
```diff
@@ -1,34 +1,33 @@
...
 
     it "return the player's tag" do
-      expect(player1.tag).to eql("x")
+      expect(player1.tag).to eql('x')
     end
 
-    it "return if the player is winner or not" do
+    it 'return if the player is winner or not' do
       expect(player1.is_winner).to eql(false)
     end
```
Comment (bertil291utn) [inline spec/player_spec.rb]: ## Positive scenario

- Specs cover positive and negative scenarios ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/Ruby_testing/commonsense.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/59566098/))
Example: there is a spec for correct usage of the method [e.g. it "creates an object if all required attributes are passed"]
and also there is spec for the wrong usage of a method  [e.g. it "does not create the object if required attributes are missing"].

  - [ ] Thus, please add positive scenario

Example:

```ruby
#positive scenario
  it 'returns true if the player is not the winner'' do
#negative scenario
  it 'returns false if the player is not the winner' do
```

**Review by bertil291utn — CHANGES_REQUESTED** [review]
# **Status:** Changes requested 🟠 
<p align="center"> image <br><strong>Not yet</strong> </p>

# Hey guys, you've been doing a great job, and also I hope you enjoyed coding.

### Your work is amazing you master it GitHub Flow you have a professional README file. 

However, in order to accomplish with all the requirements, you required to do these changes, listed below:

- [x] Edit this pull request message from "Milestone 5: Testing Tic Tac Toe Ruby code with Rspec " to "Testing Tic Tac Toe Ruby code with Rspec " due to this is not  a milestone anymore
- [x] Check to remove and adding again `.github/workflows` directory, it's not working
- [x] Please elaborate the running tests section on README file, follow this [example](https://github.com/Bekhzod96/tic_tac_toe#testing)

### OPTIONAL
 - Using require_relative for files in your directory instead of require is a ruby convention.

>"Require should be used for external files, like gems, while require_relative should be used for referring to files within your directory."

### Summary
- Remove inline comments
- One positive scenario is missing
-  Don't forget to work with inline comments below this comment box.

***

Guys work in the requested changes above and request another review.
Good luck. 

> “Never stop, never conform until what is good becomes even better, and what’s better becomes excellent.”

File: spec/game_spec.rb (hunk 1 above)
Comment (RaminMammadzada) [reply spec/game_spec.rb]: We moved ```gets.chomp``` from game.rb to main.rb.

File: spec/board_spec.rb (hunk 2 above)
Comment (RaminMammadzada) [reply spec/board_spec.rb]: We removed them.

File: spec/board_spec.rb (hunk 3 above)
Comment (RaminMammadzada) [reply spec/board_spec.rb]: We removed them too.

**Review by maelfosso — CHANGES_REQUESTED** [review]
# Congratulations :clap: 

Thank you for applying the suggested changes. :+1: 
You can merge it now :smile_cat: 

Good continuation :handshake: 

**PS**: Kindly update your README by replacing `ruby tests.rb` by `rspec`

Mael FOSSO

**Pull request merged**